src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -215,12 +216,17 @@ public void Dispose()
                 if (_valueHasBeenSet)
                 {
                     _notifyCacheEntryCommit(this);
-                    PropagateOptions(CacheEntryHelper.Current);
+
+                    if (CanPropagateOptions())
```

Comment:
Always leave a comment about why you're doing what appears to be a redundant check.